src/csharp/Grpc.Core/Internal/NativeExtension.cs

```diff
@@ -174,7 +185,7 @@ private static string GetNativeLibraryFilename()
             }
             if (PlatformApis.IsMacOSX)
             {
-                return string.Format("libgrpc_csharp_ext.{0}.dylib", architecture);
+                return string.Format("libgrpc_csharp_ext.{0}.bundle", architecture);
```

Comment:
See this topic: https://answers.unity.com/questions/23615/how-to-make-unity-find-dylib-files.html